src/telemetry_poller_vm.erl

```diff
@@ -0,0 +1,41 @@
+-module(telemetry_poller_vm).
+
+-export([
+  memory/0,
+  total_run_queue_lengths/0
+]).
+
+-spec memory() -> ok.
+memory() ->
+  Measurements = erlang:memory(),
+  telemetry:execute([vm, memory], maps:from_list(Measurements), #{}).
+
+-spec total_run_queue_lengths() -> ok.
+total_run_queue_lengths() ->
+  Total = cpu_stats(total),
+  CPU = cpu_stats(cpu),
+  telemetry:execute([vm, total_run_queue_lengths], #{
+    total => Total,
+    cpu => CPU,
+    io => Total - CPU},
+    #{}).
+
+-ifdef(OTP_RELEASE).
+  -if(?OTP_RELEASE >= 20).
+    -spec cpu_stats(total | cpu) -> non_neg_integer().
+    cpu_stats(total) ->
+      erlang:statistics(total_run_queue_lengths_all);
+    cpu_stats(cpu) ->
+      erlang:statistics(total_run_queue_lengths).
+  -else.
+    -spec cpu_stats(total | cpu) -> non_neg_integer().
+    cpu_stats(_) ->
+      lists:sum(erlang:statistics(run_queue_lengths)).
+  -endif.
+-else.
+  % OTP_RELEASE macro is available in OTP 19+ so this covers 18
+  %% TODO: Remove when support is dropped for OTP 18
```

Comment:
You know what. There has been a recent discussion about dropping OTP 18 on telemetry too, we may as well go ahead with it. What do you think about removing this branch then, updating CI and supporting only OTP 19+? I will update telemetry accordingly.

Reply:
That sounds very reasonable to me.

Reply:
@josevalim I was able to remove a few more checks at the top of the file which are also in telemetry.